inspect_image_test.go

```diff
@@ -199,42 +199,181 @@ func testInspectImage(t *testing.T, when spec.G, it spec.S) {
+      "args": ["opt", "1"],
+      "direct": true
+    }
+  ]
+}`,
+						))
+
 						info, err := subject.InspectImage("some/image", useDaemon)
 						h.AssertNil(t, err)
 
 						h.AssertEq(t, info.Processes,
 							ProcessDetails{
-								DefaultProcess: &launch.Process{
-									Type:    "other-process",
-									Command: "/other/process",
-									Args:    []string{"opt", "1"},
-									Direct:  true,
-								},
+								DefaultProcess: nil,
 								OtherProcesses: []launch.Process{
 									{
+										Type:    "other-process",
+										Command: "/other/process",
+										Args:    []string{"opt", "1"},
+										Direct:  true,
+									},
+								},
+							},
+						)
+					})
+				})
+
+				when("Platform API >= 0.4", func() {
+					when("CNB_PROCESS_TYPE is set", func() {
+						it.Before(func() {
+							h.AssertNil(t, fakeImage.SetEnv("CNB_PROCESS_TYPE", "other-process"))
+							h.AssertNil(t, fakeImage.SetEnv(PlatformAPIEnv, "0.4"))
+						})
+
+						it("ignores it and setting the correct default process", func() {
+							info, err := subject.InspectImage("some/image", useDaemon)
+							h.AssertNil(t, err)
+
+							h.AssertEq(t, info.Processes,
+								ProcessDetails{
+									DefaultProcess: &launch.Process{
 										Type:    "web",
 										Command: "/start/web-process",
 										Args:    []string{"-p", "1234"},
 										Direct:  false,
 									},
+									OtherProcesses: []launch.Process{
+										{
+											Type:    "other-process",
+											Command: "/other/process",
+											Args:    []string{"opt", "1"},
+											Direct:  true,
+										},
+									},
 								},
-							},
-						)
+							)
+						})
 					})
-				})
 
-				when("CNB_PROCESS_TYPE is set, but doesn't match an existing process", func() {
-					it.Before(func() {
-						h.AssertNil(t, fakeImage.SetEnv("CNB_PROCESS_TYPE", "missing-process"))
+					when("CNB_PROCESS_TYPE is set, but doesn't match an existing process", func() {
+						it.Before(func() {
+							h.AssertNil(t, fakeImage.SetEnv("CNB_PROCESS_TYPE", "missing-process"))
+							h.AssertNil(t, fakeImage.SetEnv(PlatformAPIEnv, "0.4"))
+						})
+
+						it("returns web as default default process", func() {
+							info, err := subject.InspectImage("some/image", useDaemon)
+							h.AssertNil(t, err)
+
+							h.AssertEq(t, info.Processes,
+								ProcessDetails{
+									DefaultProcess: &launch.Process{
+										Type:    "web",
+										Command: "/start/web-process",
+										Args:    []string{"-p", "1234"},
+										Direct:  false,
+									},
+									OtherProcesses: []launch.Process{
+										{
+											Type:    "other-process",
+											Command: "/other/process",
+											Args:    []string{"opt", "1"},
+											Direct:  true,
+										},
+									},
+								},
+							)
+						})
 					})
 
-					it("returns a nil default process", func() {
+					it("returns a nil default process when there is no web process", func() {
```

Comment:
This doesn't seem correct to me :/

Reply:
Looking here: https://github.com/buildpacks/rfcs/blob/main/text/0045-launcher-arguments.md 

Specifically here: 

> The exporter will prepend /cnb/process to the image PATH. The exporter will prepend /cnb/lifecycle to the image PATH. The exporter will set the app image entrypoint to the symlink matching the default process.

But, perhaps I'm misunderstanding something. I'm not very familiar with how it worked before.

Reply:
I'm also not. I just lifted that test from the pre Platform API 0.4, but I don't know if there was a change re: to the above.

Reply:
Could you elaborate on what doesn't seem right?

I'm looking at [this](https://github.com/buildpacks/spec/blob/platform/v0.4/platform.md#outputs-4):

```
- If <process-type> matches a buildpack-provided process:
    - MUST have ENTRYPOINT=/cnb/process/<process-type>
- If <process-type> does not match a buildpack-provided process:
    - If there is exactly one buildpack-provided process:
        - MUST have ENTRYPOINT=/cnb/process/<type> where <type> matches the type of the process
           Else:
         -MUST have ENTRYPOINT set to /cnb/lifecycle/launcher
```

which leads me to believe that there is a potential of having no default process type.

Reply:
I agree. I guess my point is that we shouldn't be looking at `CNB_PROCESS_TYPE` in the app image to determine the default process type. I don't think the exporter sets that for platform API >= 0.4: https://github.com/buildpacks/lifecycle/blob/f65216e2ad4692027fb64649fb1187a1b502e8be/exporter_test.go#L1021

We should be looking at the entrypoint. I don't see that in this PR...

Reply:
I now use the entrypoint! Let me know if this looks more correct